src/azure/CodeGeneratorCsa.cs

```diff
@@ -70,6 +70,13 @@ public class CodeGeneratorCsa : CodeGeneratorCs
                 var exceptionTemplate = new ExceptionTemplate {Model = exceptionType};
                 await Write(exceptionTemplate,
                      $"{GeneratedSourcesBaseFolder}{FolderModels}/{exceptionTemplate.Model.ExceptionTypeDefinitionName}{ImplementationFileExtension}");
+                
+                if(!existingModelTypes.Contains(exceptionType))
+                {
+                    await Write(new ModelTemplate{ Model = exceptionType },

```

Comment:
wouldn't this also happen if you just called `GenerateModels` (line 42) with `codeModel.ErrorTypes` unioned in as well? Or is there a difference to how this writes the new model compared to `GenerateModels`?